sdk/servicebus/service-bus/test/internal/receiver.spec.ts

```diff
@@ -0,0 +1,249 @@
+            (receiver as any).close = () => {
+              closeWasCalled = true;
+            };
+          }
+        }),
+        "peekLock"
+      );
+
+      const subscription = await subscribeAndWaitForInitialize(receiverImpl);
+      const originalStreamingReceiver = receiverImpl["_context"].streamingReceiver;
+
+      await subscription.close();
+
+      assert.isFalse(
+        closeWasCalled,
+        "closing a subscription does NOT close the receiver (it should be re-usable)"
+      );
+
+      const subscription2 = await subscribeAndWaitForInitialize(receiverImpl);
+
+      assert.equal(
+        originalStreamingReceiver?.name,
+        receiverImpl["_context"].streamingReceiver?.name,
+        "StreamingReceiver is closed but not replaced - this allows us to just stop and start at will without losing anything."
+      );
+
+      await subscription2.close();
+
+      await receiverImpl.close();
+    });
+
+    it("can re-subscribe after previous subscription is aborted", async () => {
+      const receiverImpl = new ReceiverImpl(createClientEntityContextForTests(), "peekLock");
+
+      const abortSignal = {
+        aborted: true
+      } as AbortSignalLike;
+
+      const subscription = receiverImpl.subscribe(
+        {
+          processError: async (_err) => {},
+          processMessage: async (_msg) => {}
+        },
+        {
+          //
+          // TODO: why is tracingOptions mandatory now?
+          //
+          tracingOptions: {},
+          abortSignal
+        }
+      );
+
+      // subscription should just be auto-closed already
+      const subscription2 = receiverImpl.subscribe({
+        processError: async (_err) => {},
+        processMessage: async (_msg) => {}
+      });
+
+      await subscription.close(); // and closing it "out of order" shouldn't be an issue either.
+      await subscription2.close();
+      await receiverImpl.close();
+    });
+
+    async function subscribeAndWaitForInitialize<
+      T extends ReceivedMessage | ReceivedMessageWithLock
+    >(receiver: ReceiverImpl<T>): Promise<ReturnType<typeof receiver["subscribe"]>> {
+      const sub = await new Promise<{
+        close(): Promise<void>;
+      }>((resolve, reject) => {
+        const subscription = receiver.subscribe({
+          processInitialize: async () => {
+            resolve(subscription);
+          },
+          processError: async (err) => {
+            reject(err);
+          },
+          processMessage: async (_msg) => {}
+        } as InternalMessageHandlers<any>);
+      });
+
+      assert.exists(
+        receiver["_context"].streamingReceiver,
+        "streaming receiver has been initialized in the context"
+      );
+
+      assert.isTrue(
+        receiver["_context"].streamingReceiver?.isReceivingMessages,
+        "streaming receiver should indicate it's receiving messages"
+      );
+
+      return sub;
+    }
+
+    // function assertProperListenerCleanup(receiver: Receiver): void {
```

Comment:
Did you mean to remove this?

Reply:
I did, it turned out to be an incorrect assumption on my part :)